### PR TITLE
[Heartbeat] filter dev flags inside synthetics args

### PR DESCRIPTION
**Pull request by emilioalvap**
## What does this PR do?

Closes https://github.com/elastic/beats/issues/35050.

Filter dev-related flags specified in `synthetics_args` when monitor comes from ui.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/heartbeat/monitors/browser/project.go [hunk 1]
```diff
@@ -140,17 +160,19 @@ func (p *Project) jobs() []jobs.Job {
 
 	isScript := p.projectCfg.Source.Inline != nil
 	ctx := context.WithValue(p.ctx, synthexec.SynthexecTimeout, p.projectCfg.Timeout+30*time.Second)
+	sFields := p.StdFields()
 
 	if isScript {
 		src := p.projectCfg.Source.Inline.Script
-		j = synthexec.InlineJourneyJob(ctx, src, p.Params(), p.StdFields(), p.extraArgs()...)
+		j = synthexec.InlineJourneyJob(ctx, src, p.Params(), sFields, p.extraArgs(sFields.Origin != "")...)
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/project.go:167]: TIL: Does only project based monitors have this origin field set?

File: x-pack/heartbeat/monitors/browser/project.go [hunk 2]
```diff
@@ -167,3 +189,63 @@ func (p *Project) plugin() plugin.Plugin {
...
+
+		if !next {
+			break
+		}
+
+		if pCount, ok := filter[iter.Val()]; ok {
+		ParamsIter:
+			for i := 0; i < pCount; i++ {
+				// Found filtered flag, check if it has associated params
+				if param, ok := iter.Peek(); ok && !strings.HasPrefix(param, "-") {
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/project.go:239]: we have a variadic options for some of the params like `tags` which can spawn multiple values, We would need to handle that as well. Also add a test case.

I would also drop those options from the UI origin as we don't want any level of filtering from the UI for project based monitors - https://github.com/elastic/synthetics/issues/780

File: x-pack/heartbeat/monitors/browser/project.go [hunk 3]
```diff
@@ -94,8 +95,27 @@ func (p *Project) Close() error {
 	return nil
 }
 
-func (p *Project) extraArgs() []string {
-	extraArgs := p.projectCfg.SyntheticsArgs
+var filterMap = map[string]int{
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/project.go:98]: Can you add a comment here explaining 1 specifies atleast one value for the option

File: x-pack/heartbeat/monitors/browser/project.go [hunk 4]
```diff
@@ -94,8 +95,27 @@ func (p *Project) Close() error {
...
+	"--pause-on-error":  0,
+	"--quiet-exit-code": 0,
+	"--dry-run":         0,
+	"--outfd":           1,
+	"--reporter":        1,
+	"-V":                0,
+	"--version":         0,
+	"-h":                0,
+	"--help":            0,
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/project.go:107]: couple of this to add 
* --inline - Would make it hard to debug when accidentally done 
* -r, --require - Modules to preload, this wont work well for project based ones
* --ws-endpoint, Listening to chrome over WS connection, wont work in service

**Review by vigneshshanmugam — COMMENTED** [review]
Overall the code looks good to me, pending the small changes.

File: x-pack/heartbeat/monitors/browser/project.go (hunk 1 above)
Comment (emilioalvap) [reply x-pack/heartbeat/monitors/browser/project.go:167]: Only UI monitors

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Pull request merged**